### PR TITLE
kconfig: drivers: bluetooth: clean up Kconfig dependencies 

### PR DESCRIPTION
**Pull request by thomasstenersen**
These are the changes from https://github.com/zephyrproject-rtos/zephyr/pull/13631 without the out-of-tree driver support. It cleans up Kconfig dependencies in Nordic drivers and in the Bluetooth subsystem.

### CODE REVIEW

File: subsys/bluetooth/host/Kconfig [hunk 1]
```diff
@@ -48,17 +48,19 @@ config BT_RX_BUF_LEN
...
-	default 1024 if BT_CTLR && BT_CENTRAL
-	default 640 if BT_CTLR
+	default 940 if BT_CTLR && (BT_LL_SW || BT_LL_SW_SPLIT) && NO_OPTIMIZATIONS
+	default 1024 if BT_CTLR && (BT_LL_SW || BT_LL_SW_SPLIT) && BT_CENTRAL
+	default 640 if BT_CTLR && (BT_LL_SW || BT_LL_SW_SPLIT)
 	default 512 if BT_USERCHAN
 	# Even if no driver is selected the following default is still
 	# needed e.g. for unit tests.
-	default 512
+	default 1024
```
Comment (carlescufi) [inline subsys/bluetooth/host/Kconfig]: Why did you change the default here?

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/host/Kconfig (hunk 1 above)
Comment (thomasstenersen) [reply subsys/bluetooth/host/Kconfig]: To accommodate the worst case stack requirement if an out-of-tree controller is used. I'll add a comment in the code here and in the commit with the reasoning.

**Review by aescolar — APPROVED** [review]
I was ok with the previous PR, I'm also ok with this one :)

**Review by carlescufi — APPROVED** [review]

**Review by ulfalizer — APPROVED** [review]
Re-approving

**Review by cvinayak — APPROVED** [review]

**Comment by carlescufi** [comment]
Merging since this is the non-contentious portion of #13631  which was already reviewed several days ago.
